File: Micro_Eva_Cola/cola.java
import java.util.LinkedList;
import java.util.Queue;

public class Cola {
    private Queue<Integer> cola;

    public Cola() {
        cola = new LinkedList<>();
    }

    public void encolar(int valor) {
        cola.add(valor);
    }

    public int desencolar() {
        if (cola.isEmpty()) {
            return -1;
        }
        return cola.poll();
    }

    public int peek() {
        if (cola.isEmpty()) {
            return -1;
        }
        return cola.peek();
    }

    public void mostrar() {
        System.out.println(cola);
    }

    public boolean estaVacia() {
        return cola.isEmpty();
    }
}

File: MiProyecto/Implementaciones/Cola.cs
using System;
using System.Collections.Generic;
using MiProyectoTAD.Interfaces;

namespace MiProyectoTAD.Implementaciones
{
    public class Cola<T> : ICola<T>
    {
        private Queue<T> elementos = new Queue<T>();

        public void Encolar(T elemento)
        {
            elementos.Enqueue(elemento);
        }

        public T Desencolar()
        {
            if (EstaVacia())
                throw new InvalidOperationException("Cola vacía");

            return elementos.Dequeue();
        }

        public T Peek()
        {
            if (EstaVacia())
                throw new InvalidOperationException("Cola vacía");

            return elementos.Peek();
        }

        public bool EstaVacia()
        {
            return elementos.Count == 0;
        }
    }
}